api/envoy/config/core/v3/protocol.proto

```diff
@@ -224,6 +224,11 @@ message HttpProtocolOptions {
   // <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.HttpConnectionManager.drain_timeout>`.
   google.protobuf.Duration max_connection_duration = 3;
 
+  // The maximum response headers size allowed from the upstream server. If not set, the default max response headers
+  // size allowed is 80 KiB. Response that exceed this limit would fail with a protocol error.
+  google.protobuf.UInt32Value max_response_headers_kb = 7
```

Comment:
@htuch I'm wondering if this field should be here or in the HCM (similar to [`max_request_headers_kb`](https://github.com/envoyproxy/envoy/blob/4632a68618322c482e3b750adf0805ed9c01460a/api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto#L439)). I tend to add it to HttpProtocolOptions as suggested, but would like to know if you think otherwise.

Reply:
I think it's fine here - it's arguably a property of the upstream so not HCM. That said, maybe HCM should migrate its request header sizes here for consistency.